python/lsst/ts/scriptqueue/ui/request_cmd.py

```diff
@@ -0,0 +1,385 @@
+__all__ = ["RequestCmd"]
+
+import logging
+import warnings
+import argparse
+import yaml
+
+from lsst.ts.scriptqueue.utils import DETAIL_LEVEL
+try:
+    from cmd2 import Cmd
+except ModuleNotFoundError as e:
+    warnings.warn("Could not find module cmd2. Fallback to standard "
+                  "cmd library. This may limit the functionality of the shell.")
+    from cmd import Cmd
+
+from .request_model import RequestModel
+
+queue_names = ["Main", "Auxiliary"]
+
+
+class RequestCmd(Cmd):
+    """This class provides a command line interface to the RequestModel class.
+    """
+
+    def __init__(self, index):
+
+        self.intro = """Welcome to the Request Script shell. This script is designed to
+        operate with the LSST script queue.
+
+        Type help or ? to list commands."""
+        self.prompt = '(cmd): '
+
+        Cmd.__init__(self)
+
+        self.log = logging.getLogger("request")
+        self.model = RequestModel(index)
+
+        self.__run_parser = argparse.ArgumentParser(prog="run")
+        self.__run_parser.add_argument('-e', '--external', help="Run an external script.")
+        self.__run_parser.add_argument('-s', '--standard', help="Run a standard script.")
+        self.__run_parser.add_argument('-c', '--config', help="Configuration file for the script.")
+        self.__run_parser.add_argument('-p', '--parameters', nargs='+',
+                                       help="In line configuration string for the script. Must be "
+                                            "in yaml format.")
+        self.__run_parser.add_argument("--monitor", dest="monitor", action="store_true",
+                                       help="Monitor script execution all the way to the end.")
+        self.__run_parser.add_argument("--timeout", dest="timeout", default=120., type=float,
+                                       help="Timeout for monitoring scripts.")
+
+        self.__set_log_parser = argparse.ArgumentParser(prog="log_parser")
+        self.__set_log_parser.add_argument("-v", "--verbose", dest="verbose",
+                                           action='count', default=0,
+                                           help="Set the verbosity for the console logging.")
+
+        self.__set_script_log_parser = argparse.ArgumentParser(prog="script_log")
+        self.__set_script_log_parser.add_argument("-v", "--verbose", dest="verbose",
+                                                  action='count',
+                                                  default=0,
+                                                  help="Set the verbosity for the console logging.")
```

Comment:
Why is this optional? I would expect the user to always need to specify a log level and perhaps always the index. I'd be tempted to simplify this by:
- removing this parser
- require that the arguments to do_set_script_log_level be a pair of integers (easily enforced without argparse, e.g.:
```
try:
   arglist = args.split()
   assert len(arglist) == 2
   level, index = [int(arg) for arg in args]
except Exception:
    print(f"*** Could not parse {args!r} as two integers")
```
It would save trouble to have verbosity an argument to "run", but that either requires some asynchronous programming in here or a new argument "logLevel" to the script queue's "add" command.